pkg/api/v1/pod/util.go

```diff
@@ -301,6 +301,16 @@ func IsPodReady(pod *v1.Pod) bool {
 	return IsPodReadyConditionTrue(pod.Status)
 }
 
+// IsPodTerminal returns true if a pod is terminal, all containers are stopped and cannot ever regress.
+func IsPodTerminal(pod *v1.Pod) bool {
+	return IsPodPhaseTerminal(pod.Status.Phase)
```

Comment:
Maybe need a nil check for pod before this line?

```suggestion
	return pod != nil && IsPodPhaseTerminal(pod.Status.Phase)
```